rust/crd/src/authentication.rs

```diff
@@ -323,15 +324,15 @@ async fn check_or_generate_admin_credentials(
     auto_generate: &bool,
 ) -> Result<bool, Error> {
     match client
-        .exists::<Secret>(secret_name, Some(secret_namespace))
+        .get_opt::<Secret>(secret_name, Some(secret_namespace))
         .await
         .with_context(|_| KubeSnafu {
             reason: format!(
                 "checking if admin credential secret exists [{}/{}]",
                 secret_name, secret_namespace
             ),
         })? {
-        true => {
+        Some(_) => {
             // The secret exists, retrieve the content and check that all required keys are present
             // any missing keys will be filled with default or generated values
             let secret_content: Secret = client
```

Comment:
No need to refetch the secret from the API when you just got (and discarded) it from `get_opt`

Reply:
you are right indeed. Shouldn't searchreplace these changes :)